packages/core/src/Machine.ts

```diff
@@ -50,23 +51,26 @@ export function Machine<
 export function createMachine<
   TContext,
   TEvent extends EventObject = AnyEventObject,
-  TTypestate extends Typestate<TContext> = { value: any; context: TContext }
+  TTypestate extends Typestate<TContext> = { value: any; context: TContext },
+  TMeta extends TypegenMeta = DefaultTypegenMeta
 >(
   config: TContext extends Model<any, any, any, any>
     ? 'Model type no longer supported as generic type. Please use `model.createMachine(...)` instead.'
-    : MachineConfig<TContext, any, TEvent>,
-  options?: Partial<MachineOptions<TContext, TEvent>>
-): StateMachine<TContext, any, TEvent, TTypestate>;
+    : MachineConfig<TContext, any, TEvent> & { types?: TMeta },
```

Comment:
q: why don't we just put `types?: Meta` in the `MachineConfig`?

Reply:
We could do, but frankly I couldn't be bothered to go through and add all the generics. If this is the only entry point (at the root of the config), we should only allow it to be passed in here.

Reply:
And `MachineConfig` only represents the root so it IMHO makes sense for it to be included there. This also improves some minor things like jump to type declaration, gets displayed better in tooltips and stuff.

Reply:
OK, will add it there

Reply:
I've just added it, but there's one thing on my mind... Let's say that a user a defines their machine config, typed like this, outside of the machine call:

```ts
const config: MachineConfig<Context, any, Event> = {};
```

They'd now have to change the declaration to this:

```ts
const config: MachineConfig<Context, any, Event, any, Meta> = {};
```

Which is not particularly fun.

Reply:
Is it a problem though? you have added a default type param to `MachineConfig`

Reply:
I mean they'd need to do it if they wanted the inference to carry over to the machine. But maybe we just don't mind that much.

Reply:
We'd have to see what would happen with:
```js
const config: MachineConfig<Context, any, Event> = {};
createMachine({ ...config, types: {} as Meta })
```
but either way, I think it's OK - I don't think the `Meta` should be used manually anyway because that's just asking for problems